packages/edit-post/src/components/sidebar/post-template/hook.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { store as coreStore } from '@wordpress/core-data';
+import { useMemo, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as editPostStore } from '../../../store';
+
+export default function usePostTemplateForm() {
+	const isViewable = useSelect( ( select ) => {
+		const { getCurrentPostType } = select( editorStore );
+		const { getPostType } = select( coreStore );
+		return getPostType( getCurrentPostType() )?.viewable ?? false;
+	}, [] );
+
+	const settings = useSelect(
+		( select ) => select( editorStore ).getEditorSettings(),
+		[]
+	);
+
+	const canUserCreateTemplate = useSelect(
+		( select ) => select( coreStore ).canUser( 'create', 'templates' ),
+		[]
+	);
+
+	const currentPostId = useSelect(
+		( select ) => select( editorStore ).getCurrentPostId(),
+		[]
+	);
+
+	const templates = useSelect( ( select ) => {
+		const { getCurrentPostType } = select( editorStore );
+		const { getEntityRecords } = select( coreStore );
+		return getEntityRecords( 'postType', 'wp_template', {
+			post_type: getCurrentPostType(),
+			per_page: -1,
+		} );
+	}, [] );
+
+	const templateAttribute = useSelect(
+		( select ) =>
+			select( editorStore ).getEditedPostAttribute( 'template' ),
+		[]
+	);
+
+	const template = useSelect(
+		( select ) => select( editPostStore ).getEditedPostTemplate(),
+		[]
+	);
+
+	const { editPost } = useDispatch( editorStore );
+
+	const { __unstableSwitchToTemplateMode } = useDispatch( editPostStore );
+
+	const isVisible =
+		isViewable &&
+		( settings.availableTemplates?.length ||
+			( settings.supportsTemplateMode && canUserCreateTemplate ) );
+
+	const isPostsPage = currentPostId === settings?.page_for_posts;
```

Comment:
I think should be based on site settings and not the Editors - `const settings = getEntityRecord( 'root', 'site' );` .

Reply:
Is there a benefit to this approach? The nice thing about using editor settings is that we already have the settings object right there because we need it for `availableTemplates`.

Reply:
The `page_for_posts` isn't available in [Editor settings](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/edit-form-blocks.php#L190-L210).

Reply:
Aha! Gotcha 😀